### ISSUE TITLE
spack dependents --installed only shows one installed variant of dependent specs

### ISSUE THREAD

**Issue by omor1** [LLNL/spack#23569]
I have multiple variants of `dplasma` installed that depend on `amdlibflame/nyqwya2`. Running `spack dependents --installed amdlibflame/nyqwya2` only shows one of those installed specs.

### Steps to reproduce the issue

```console
# install multiple variants of a package that ^lapack; dplasma isn't in base, so I'll show lammps
$ spack install lammps +openmp ^amdlibflame/nyqwya2
$ spack install lammps ~openmp ^amdlibflame/nyqwya2
# check for dependents on amdliflame/nyqwya2
$ spack dependents --installed amdlibflame/nyqwya2
==> Dependents of amdlibflame@3.0%aocc@3.0.0/nyqwya2
-- linux-centos8-zen2 / aocc@3.0.0 ------------------------------
abcdefg lammps@20210310
```

Additionally, `spack find ^amdlibflame/nyqwya2` doesn't find _any_ packages that depend on that installation of `amdlibflame`. This possibly has something to do with the dependency being on the virtual `lapack` package rather than on `amdlibflame` directly.

### Error Message

```console
$ spack --debug --stacktrace dependents --installed amdlibflame/nyqwya2
lib/spack/spack/cmd/__init__.py:122 ==> [2021-05-11-11:36:09.882687] Imported dependents from built-in commands
lib/spack/spack/cmd/__init__.py:122 ==> [2021-05-11-11:36:09.883461] Imported dependents from built-in commands
lib/spack/spack/config.py:981 ==> [2021-05-11-11:36:09.908493] Reading config file /home/omor1/spack/etc/spack/defaults/config.yaml
lib/spack/spack/config.py:981 ==> [2021-05-11-11:36:09.927255] Reading config file /home/omor1/.spack/config.yaml
lib/spack/spack/database.py:363 ==> [2021-05-11-11:36:09.929759] DATABASE LOCK TIMEOUT: 3s
lib/spack/spack/database.py:367 ==> [2021-05-11-11:36:09.929939] PACKAGE LOCK TIMEOUT: No timeout
lib/spack/spack/config.py:981 ==> [2021-05-11-11:36:09.942313] Reading config file /home/omor1/spack/etc/spack/defaults/repos.yaml
lib/spack/spack/config.py:981 ==> [2021-05-11-11:36:09.944196] Reading config file /home/omor1/.spack/repos.yaml
lib/spack/spack/cmd/dependents.py:90 ==> [2021-05-11-11:36:10.021039] Dependents of amdlibflame@3.0%aocc@3.0.0/nyqwya2
-- linux-centos8-zen2 / aocc@3.0.0 ------------------------------
o4t3igp dplasma@local
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.16.1-2445-5e8b8b883c
* **Python:** 3.9.4
* **Platform:** linux-centos8-zen2
* **Concretizer:** original
```

### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

**Comment by omor1** [comment]
@tgamblin asked me to upload my installation [db tarball](https://github.com/spack/spack/files/6462236/spack-db.private_tmp.5e8b8b883c.2021-05-11-150056.tar.gz).

**Comment by alalazo** [comment]
Duplicate of #11983

**Comment by alalazo** [comment]
I think this will be fixed by #21683 once merged, can you confim?

**Comment by alalazo** [comment]
Fixed by #28673
